libflux/flux/FLUXDOC.md

```diff
@@ -0,0 +1,325 @@
+# Flux Doc Comments
+
+In the interest of establishing a single source of authority for Flux documentation,
+Flux will soon support doc comments in `stdlib`. Work is underway to integrate the
+documentation generated from doc comments into the `flux` crate, and expose them
+as JSON data for broader consumption. This document defines the details of how
+Flux doc comments should be written and formatted.
+
+The end goals of this project are as follows:
+
+1. Continuously populate the Flux documentation search bar in the InfluxDB data
+   explorer with the latest version of the docs.
+2. Expose documentation via the the Flux Language Server Protocol.
+3. Use the docs to generate a static site that will become the new home of the official Flux documentation.
+
+## Design Considerations
+
+The generated Flux docs will need to support different formats for different consumers.
+As an example, the Flux LSP and the InfluxDB UI will need to produce a condensed 
+version of the documentation for any given identifier in the Flux standard library.
+In contrast, the official docs site will need to present much more detailed information
+to readers. Consumers should be able to decide which format they want to use by
+selecting only the parts of the docs that are required for their use case.
+
+To allow for this, Flux doc comments are formatted so that the doc comment parser
+can easily distinguish the short-form content from the long-form content. In most
+cases throughout this document, these will be referred to as the "headline" and
+the "description", respectively.
+
+## Writing Flux Doc Comments
+
+Flux will treat any uninterrupted series of line comments immediately preceding
+a function definition, package clause, or option declaration, as a doc comment.
```

Comment:
This should read:

> Flux will treat any uninterrupted series of line comments immediately preceding
an assignment statement, package clause, or option declaration, as a doc comment.

Packages do not exclusively define functions but can define values of any type.